packages/admin-ui/src/filters/filter.tsx

```diff
@@ -1,17 +1,28 @@
 import React from 'react'
 import { tag } from '@vtex/admin-ui-react'
 
-import type { UseFilterReturn } from './filter.state'
-import { BaseFilter } from './base-filter'
+import { BaseFilter } from './filter-base'
+import { ComboboxItem } from '../combobox/combobox-item'
+import { focusVisible, style } from '@vtex/admin-ui-core'
+
+import type { UseFilterStateReturn } from './filter.state'
 import { FilterRadio } from './filter-radio'
-import { FilterOption } from './filter-option'
+import type { FilterItem } from '.'
+
+export const itemStyle = style({
+  display: 'flex',
+  cursor: 'pointer',
+  ...focusVisible('main'),
+})
 
-export function Filter(props: FilterProps) {
+export function Filter<T extends FilterItem>(props: FilterProps<T>) {
   const {
-    state: { listState, appliedItem },
+    state: { combobox, appliedItem, items },
     state,
   } = props
 
+  const currentSelectedId = combobox.selectedItem?.id
```

Comment:
Here, could make sense to have a default value:
```suggestion
  const currentSelectedId = combobox.selectedItem?.id ?? 0
```

Reply:
Since I'll be comparing an unknown value to this var i'd rather have it as undefined than 0